src/main/scala/nl.knaw.dans.easy.bag2deposit/package.scala

```diff
@@ -55,13 +56,212 @@ package object bag2deposit extends DebugEnhancedLogging {
+      //    "<e2><80><b9>" -> "‹",
+      //    "<e2><80><ba>" -> "›",
+      //    "<e2><80><bb>" -> "※",
+      //    "<e2><80><bc>" -> "‼",
+      //    "<e2><80><bd>" -> "‽",
+      //    "<e2><80><be>" -> "‾",
+      //    "<e2><80><bf>" -> "‿",
+      //    "<e2><81><80>" -> "⁀",
+      //    "<e2><81><81>" -> "⁁",
+      //    "<e2><81><82>" -> "⁂",
+      //    "<e2><81><83>" -> "⁃",
+      //    "<e2><81><84>" -> "⁄",
+      //    "<e2><81><85>" -> "⁅",
+      //    "<e2><81><86>" -> "⁆",
+      //    "<e2><81><87>" -> "⁇",
+      //    "<e2><81><88>" -> "⁈",
+      //    "<e2><81><89>" -> "⁉",
+      //    "<e2><81><8a>" -> "⁊",
+      //    "<e2><81><8b>" -> "⁋",
+      //    "<e2><81><8c>" -> "⁌",
+      //    "<e2><81><8d>" -> "⁍",
+      //    "<e2><81><8e>" -> "⁎",
+      //    "<e2><81><8f>" -> "⁏",
+      //    "<e2><81><90>" -> "⁐",
+      //    "<e2><81><91>" -> "⁑",
+      //    "<e2><81><92>" -> "⁒",
+      //    "<e2><81><93>" -> "⁓",
+      //    "<e2><81><94>" -> "⁔",
+      //    "<e2><81><95>" -> "⁕",
+      //    "<e2><81><96>" -> "⁖",
+      //    "<e2><81><97>" -> "⁗",
+      //    "<e2><81><98>" -> "⁘",
+      //    "<e2><81><99>" -> "⁙",
+      //    "<e2><81><9a>" -> "⁚",
+      //    "<e2><81><9b>" -> "⁛",
+      //    "<e2><81><9c>" -> "⁜",
+      //    "<e2><81><9d>" -> "⁝",
+      //    "<e2><81><9e>" -> "⁞",
+      //    "<e2><81><b0>" -> "⁰",
+      //    "<e2><81><b1>" -> "ⁱ",
+      //    "<e2><81><b4>" -> "⁴",
+      //    "<e2><81><b5>" -> "⁵",
+      //    "<e2><81><b6>" -> "⁶",
+      //    "<e2><81><b7>" -> "⁷",
+      //    "<e2><81><b8>" -> "⁸",
+      //    "<e2><81><b9>" -> "⁹",
+      //    "<e2><81><ba>" -> "⁺",
+      //    "<e2><81><bb>" -> "⁻",
+      //    "<e2><81><bc>" -> "⁼",
+      //    "<e2><81><bd>" -> "⁽",
+      //    "<e2><81><be>" -> "⁾",
+      //    "<e2><81><bf>" -> "ⁿ",
+    )
+    lower.toSeq ++ lower.map(kv => kv._1.toUpperCase -> kv._2).toSeq
+  }
+
   def loadXml(file: File): Try[Elem] = {
     trace(file)
-    Try(XML.loadFile(file.toJava))
-      .recoverWith {
-        case t: FileNotFoundException => Failure(InvalidBagException(s"could not find: $file"))
-        case t: SAXParseException => Failure(InvalidBagException(s"could not load: $file - ${ t.getMessage }"))
-      }
+    Try {
+      val withoutPrologue = file.contentAsString.replaceAll("<[?].+[?]>", "")
+      val string = // make sure to have a prologue
+        """<?xml version="1.0" encoding="UTF-8" ?>
+          |""".stripMargin + encodingMap.foldLeft(withoutPrologue)((acc, kv) =>
+          acc.replaceAll(kv._1, kv._2)
+        )
+      if (string.matches("(?s).*<..><..>.*"))
+        throw new InvalidBagException("DDM contains invalid encoding with the style <..><..>")
+      // without this extra scan of the document we would get
+      // InvalidBagException: Could not load: PATH-TO/dataset.xml - The content of elements must consist of well-formed character data or markup.
```

Comment:
With this scan also in advance we could prevent the multitude of other scans in case of a significant performance penalty.